qoffer-domain/src/main/groovy/life/qbic/portal/portlet/offers/create/CreateOffer.groovy

```diff
@@ -27,17 +27,19 @@ class CreateOffer implements CreateOfferInput{
     @Override
     void createOffer(Offer offerContent) {
         OfferId identifier = generateQuotationID(offerContent.customer)
+        double offerPrice = PriceCalculator.calculateOfferPrice(offerContent.items,offerContent.selectedCustomerAffiliation.category)
 
-        Offer finalizedOffer = new Offer.Builder(new Date(),
-                offerContent.expirationDate,
+        Offer finalizedOffer = new Offer.Builder(
                 offerContent.customer,
                 offerContent.projectManager,
-                offerContent.projectDescription,
                 offerContent.projectTitle,
+                offerContent.projectDescription,
                 offerContent.items,
-                calculateOfferPrice(offerContent.items,offerContent.selectedCustomerAffiliation.category),
-                identifier,
                 offerContent.selectedCustomerAffiliation)
+                .identifier(identifier)
+                .expirationDate(new Date(2030,12,24)) //todo how to determine this?
+                .modificationDate(new Date())
+                .totalPrice(offerPrice)
```

Comment:
The total price should be computed by the object internally that holds the business rule and not passed as already calculated value. This violates the OOP approach of business rules.

Reply:
Should I refactor it in this PR or should this be done in another PR with a respecticve card?